src/screens/Login/ConfirmMnemonic/ConfirmMnemonicContainer.js

```diff
@@ -5,47 +5,56 @@
 
 import React, { PureComponent } from 'react'
 import { Alert } from 'react-native'
+import { connect } from 'react-redux'
+import { bindActionCreators } from 'redux'
 import PropTypes from 'prop-types'
+import { getAddress } from '@chronobank/bitcoin/utils'
+import { bitcoinSaveAddress } from '@chronobank/bitcoin/redux/actions'
+import { ethSaveAddress } from '@chronobank/ethereum/redux/actions'
+import { getPrivateKeyByMnemonic, getAddressByMnemonic } from '@chronobank/ethereum/utils'
+import { loginThunk } from '../../../redux/session/thunks'
 import { MNEMONIC_LENGTH } from '../../../common/constants/globals'
 import i18n from '../../../locales/translation'
 import ConfirmMnemonic from './ConfirmMnemonic'
 
+const mapStateToProps = (ownState, ownProps) => {
+  return {}
+}
+
+const mapDispatchToProps = (dispatch) => bindActionCreators({
+  loginThunk,
+  bitcoinSaveAddress,
+  ethSaveAddress,
+}, dispatch)
+
 class ConfirmMnemonicContainer extends PureComponent {
-  constructor (props) {
+  constructor(props) {
     super(props)
     this.state = this.createInitialState()
   }
 
-  handleDone = async () => {
+  handleDone = () => {
     const {
       mnemonic,
-      password,
     } = this.props.navigation.state.params
     const {
-      usePinProtection,
       navigation,
-      onMnemonicLogin,
-      onLogin,
+      loginThunk,
+      bitcoinSaveAddress,
+      ethSaveAddress,
     } = this.props
 
     if (mnemonic !== this.state.mnemonic.join(' ')) {
       this.addError(i18n.t('ConfirmMnemonic.wrongMnemonicError'))
       return this.resetState()
     }
-
-    const { privateKey } = await onMnemonicLogin(mnemonic) //need action from @chronobank
-
-    if (!usePinProtection) {
-      return onLogin()
-    }
-
-    const params = {
-      mnemonic,
-      password,
-      privateKey,
-    }
-
-    navigation.navigate('EnterPin', params)
+    const privateKey = getPrivateKeyByMnemonic(mnemonic)
+    const bitcoinAddress = getAddress(privateKey)
```

Comment:
This logic should be relocated src/redux/login. Containers must only call thunks  and do not contain non-UI logic.